spec/models/upload_spec.rb

```diff
@@ -156,4 +156,27 @@
       expect(upload.col_sep).to eq(',')
     end
   end
+
+  describe 'set_col_sep' do
+    it 'sets col_sep to comma when csv first line' do
+      first_line = 'a,b,c'
+      upload = create :upload
```

Comment:
You can use `build` here.  It saves us the time of persisting to the database.  Minor change in speed for one spec, but if we consistently only write to the database when we have to, it helps the speed of running the whole test suite.
Applies to the other two `upload` creations below too.

```suggestion
      upload = build :upload
```